From 278e6708507cb6cc5fe542052cb6b170e9c004a0 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 26 Oct 2017 11:14:44 +0800
Subject: [PATCH 1/4] sunxi: change the DE1 video option to CONFIG_VIDEO_SUNXI

The sunxi DE1 video option used to be CONFIG_VIDEO, which has the same
name as the "Enable legacy video support" option in
drivers/video/Kconfig.

Change the option name to CONFIG_VIDEO_SUNXI, which is really used by
Makefile under drivers/video/sunxi/, and defined in sunxi-common.h
when CONFIG_VIDEO is selected before this change. Now CONFIG_VIDEO_SUNXI
selects CONFIG_VIDEO and the usages of CONFIG_VIDEO in sunxi Kconfig and
config headers are all converted to use CONFIG_VIDEO_SUNXI.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm/mach-sunxi/Kconfig    | 29 +++++++++++++++--------------
 include/configs/sunxi-common.h |  6 ++----
 scripts/config_whitelist.txt   |  1 -
 3 files changed, 17 insertions(+), 19 deletions(-)

diff --git a/arch/arm/mach-sunxi/Kconfig b/arch/arm/mach-sunxi/Kconfig
index 2309f59999..3c29fc61f7 100644
--- a/arch/arm/mach-sunxi/Kconfig
+++ b/arch/arm/mach-sunxi/Kconfig
@@ -606,7 +606,7 @@ config AXP_GPIO
 	---help---
 	Say Y here to enable support for the gpio pins of the axp PMIC ICs.
 
-config VIDEO
+config VIDEO_SUNXI
 	bool "Enable graphical uboot console on HDMI, LCD or VGA"
 	depends on !MACH_SUN8I_A83T
 	depends on !MACH_SUNXI_H3_H5
@@ -614,6 +614,7 @@ config VIDEO
 	depends on !MACH_SUN8I_V3S
 	depends on !MACH_SUN9I
 	depends on !MACH_SUN50I
+	select VIDEO
 	default y
 	---help---
 	Say Y here to add support for using a cfb console on the HDMI, LCD
@@ -622,21 +623,21 @@ config VIDEO
 
 config VIDEO_HDMI
 	bool "HDMI output support"
-	depends on VIDEO && !MACH_SUN8I
+	depends on VIDEO_SUNXI && !MACH_SUN8I
 	default y
 	---help---
 	Say Y here to add support for outputting video over HDMI.
 
 config VIDEO_VGA
 	bool "VGA output support"
-	depends on VIDEO && (MACH_SUN4I || MACH_SUN7I)
+	depends on VIDEO_SUNXI && (MACH_SUN4I || MACH_SUN7I)
 	default n
 	---help---
 	Say Y here to add support for outputting video over VGA.
 
 config VIDEO_VGA_VIA_LCD
 	bool "VGA via LCD controller support"
-	depends on VIDEO && (MACH_SUN5I || MACH_SUN6I || MACH_SUN8I)
+	depends on VIDEO_SUNXI && (MACH_SUN5I || MACH_SUN6I || MACH_SUN8I)
 	default n
 	---help---
 	Say Y here to add support for external DACs connected to the parallel
@@ -663,14 +664,14 @@ config VIDEO_VGA_EXTERNAL_DAC_EN
 
 config VIDEO_COMPOSITE
 	bool "Composite video output support"
-	depends on VIDEO && (MACH_SUN4I || MACH_SUN5I || MACH_SUN7I)
+	depends on VIDEO_SUNXI && (MACH_SUN4I || MACH_SUN5I || MACH_SUN7I)
 	default n
 	---help---
 	Say Y here to add support for outputting composite video.
 
 config VIDEO_LCD_MODE
 	string "LCD panel timing details"
-	depends on VIDEO
+	depends on VIDEO_SUNXI
 	default ""
 	---help---
 	LCD panel timing details string, leave empty if there is no LCD panel.
@@ -680,14 +681,14 @@ config VIDEO_LCD_MODE
 
 config VIDEO_LCD_DCLK_PHASE
 	int "LCD panel display clock phase"
-	depends on VIDEO
+	depends on VIDEO_SUNXI
 	default 1
 	---help---
 	Select LCD panel display clock phase shift, range 0-3.
 
 config VIDEO_LCD_POWER
 	string "LCD panel power enable pin"
-	depends on VIDEO
+	depends on VIDEO_SUNXI
 	default ""
 	---help---
 	Set the power enable pin for the LCD panel. This takes a string in the
@@ -695,7 +696,7 @@ config VIDEO_LCD_POWER
 
 config VIDEO_LCD_RESET
 	string "LCD panel reset pin"
-	depends on VIDEO
+	depends on VIDEO_SUNXI
 	default ""
 	---help---
 	Set the reset pin for the LCD panel. This takes a string in the format
@@ -703,7 +704,7 @@ config VIDEO_LCD_RESET
 
 config VIDEO_LCD_BL_EN
 	string "LCD panel backlight enable pin"
-	depends on VIDEO
+	depends on VIDEO_SUNXI
 	default ""
 	---help---
 	Set the backlight enable pin for the LCD panel. This takes a string in the
@@ -712,7 +713,7 @@ config VIDEO_LCD_BL_EN
 
 config VIDEO_LCD_BL_PWM
 	string "LCD panel backlight pwm pin"
-	depends on VIDEO
+	depends on VIDEO_SUNXI
 	default ""
 	---help---
 	Set the backlight pwm pin for the LCD panel. This takes a string in the
@@ -720,14 +721,14 @@ config VIDEO_LCD_BL_PWM
 
 config VIDEO_LCD_BL_PWM_ACTIVE_LOW
 	bool "LCD panel backlight pwm is inverted"
-	depends on VIDEO
+	depends on VIDEO_SUNXI
 	default y
 	---help---
 	Set this if the backlight pwm output is active low.
 
 config VIDEO_LCD_PANEL_I2C
 	bool "LCD panel needs to be configured via i2c"
-	depends on VIDEO
+	depends on VIDEO_SUNXI
 	default n
 	select CMD_I2C
 	---help---
@@ -776,7 +777,7 @@ config VIDEO_DE2
 
 choice
 	prompt "LCD panel support"
-	depends on VIDEO
+	depends on VIDEO_SUNXI
 	---help---
 	Select which type of LCD panel to support.
 
diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 91751171ec..7d86b58417 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -268,7 +268,7 @@ extern int soft_i2c_gpio_scl;
 /* GPIO */
 #define CONFIG_SUNXI_GPIO
 
-#ifdef CONFIG_VIDEO
+#ifdef CONFIG_VIDEO_SUNXI
 /*
  * The amount of RAM to keep free at the top of RAM when relocating u-boot,
  * to use as framebuffer. This must be a multiple of 4096.
@@ -278,8 +278,6 @@ extern int soft_i2c_gpio_scl;
 /* Do we want to initialize a simple FB? */
 #define CONFIG_VIDEO_DT_SIMPLEFB
 
-#define CONFIG_VIDEO_SUNXI
-
 #define CONFIG_VIDEO_LOGO
 #define CONFIG_VIDEO_STD_TIMINGS
 #define CONFIG_I2C_EDID
@@ -288,7 +286,7 @@ extern int soft_i2c_gpio_scl;
 /* allow both serial and cfb console. */
 /* stop x86 thinking in cfbconsole from trying to init a pc keyboard */
 
-#endif /* CONFIG_VIDEO */
+#endif /* CONFIG_VIDEO_SUNXI */
 
 /* Ethernet support */
 #ifdef CONFIG_SUNXI_EMAC
diff --git a/scripts/config_whitelist.txt b/scripts/config_whitelist.txt
index 3037d6125b..0b00a019bc 100644
--- a/scripts/config_whitelist.txt
+++ b/scripts/config_whitelist.txt
@@ -5078,7 +5078,6 @@ CONFIG_VIDEO_MXS
 CONFIG_VIDEO_MXS_MODE_SYSTEM
 CONFIG_VIDEO_OMAP3
 CONFIG_VIDEO_STD_TIMINGS
-CONFIG_VIDEO_SUNXI
 CONFIG_VIDEO_VCXK
 CONFIG_VID_FLS_ENV
 CONFIG_VM86
-- 
2.13.6

